### DISCUSSION TITLE
Can't build Ruby 2.6.8 on Fedora 36 Docker base image

### DISCUSSION THREAD

**Discussion by nickmarden** [rbenv/ruby-build#2008] [Build failures]
### Steps to reproduce the behavior

```
$ docker run -it --rm fedora:36

[root@087d2a926d59 /]# dnf install -y gcc make bzip2 openssl-devel libyaml-devel libffi-devel readline-devel zlib-devel gdbm-devel ncurses-devel git perl-FindBin perl-File-Compare && \
  dnf groupinstall -y "C Development Tools and Libraries" "Development Tools" && \
  git clone https://github.com/rbenv/ruby-build.git && \
  cd ruby-build && \
  ./install.sh && \
  ruby-build 2.6.8 /usr
...
# Lots and lots of installing happens
...
Cloning into 'ruby-build'...
remote: Enumerating objects: 12982, done.
remote: Counting objects: 100% (1675/1675), done.
remote: Compressing objects: 100% (545/545), done.
remote: Total 12982 (delta 1329), reused 1350 (delta 1061), pack-reused 11307
Receiving objects: 100% (12982/12982), 2.64 MiB | 3.40 MiB/s, done.
Resolving deltas: 100% (8796/8796), done.
Downloading openssl-1.1.1q.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/d7939ce614029cdff0b6c20f0e2e5703158a489a72b2507b8bd51bf8c8fd10ca
Installing openssl-1.1.1q...
Installed openssl-1.1.1q to /usr

Downloading ruby-2.6.8.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.6/ruby-2.6.8.tar.bz2
Installing ruby-2.6.8...

BUILD FAILED (Fedora Linux 36 using ruby-build 20220713)

Inspect or clean up the working tree at /tmp/ruby-build.20220713212425.612.srFiOO
Results logged to /tmp/ruby-build.20220713212425.612.log

Last 10 log lines:
	from /tmp/ruby-build.20220713212425.612.srFiOO/ruby-2.6.8/lib/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /tmp/ruby-build.20220713212425.612.srFiOO/ruby-2.6.8/lib/rubygems/package.rb:44:in `<top (required)>'
	from /tmp/ruby-build.20220713212425.612.srFiOO/ruby-2.6.8/lib/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /tmp/ruby-build.20220713212425.612.srFiOO/ruby-2.6.8/lib/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /tmp/ruby-build.20220713212425.612.srFiOO/ruby-2.6.8/lib/rubygems/installer.rb:11:in `<top (required)>'
	from /tmp/ruby-build.20220713212425.612.srFiOO/ruby-2.6.8/lib/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /tmp/ruby-build.20220713212425.612.srFiOO/ruby-2.6.8/lib/rubygems/core_ext/kernel_require.rb:54:in `require'
	from ./tool/rbinstall.rb:707:in `<module:RbInstall>'
	from ./tool/rbinstall.rb:643:in `<main>'
make: *** [uncommon.mk:364: do-install-all] Error 1

```

### Expected vs. actual behavior

I would expect the build to succeed, but it failed.

### Logs

https://gist.github.com/nickmarden/61f174a2b7e85df380b059ce99403693

**Comment by nickmarden** [comment]
Note that it seems possible to remedy this as follows:

```
[root@087d2a926d59 ruby-build]# rpm --erase openssl-devel # Removes OpenSSL headers for 3.0.x
[root@087d2a926d59 ruby-build]# dnf install -y openssl1.1 openssl1.1-devel
...
[root@087d2a926d59 ruby-build]#  ruby-build 2.6.8 /usr
Downloading ruby-2.6.8.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.6/ruby-2.6.8.tar.bz2
Installing ruby-2.6.8...
Installed ruby-2.6.8 to /usr
```

But I'm not sure I should need to remove the OpenSSL 3.0.5 devel headers in this case, or install the OpenSSL 1.1 compat libs/headers. At the very least I think the docs could use an update.

**Comment by eregon** [comment]
This should work.
The error from the log is not really clear:
```
libssl.so.1.1: cannot open shared object file: No such file or directory - /tmp/ruby-build.20220713212425.612.srFiOO/ruby-2.6.8/.ext/x86_64-linux/openssl.so (LoadError)
```
Maybe it's a bug of CRuby's 2.6.8 build system which hardcodes the temp path to openssl.so.

I wonder if the `/usr` prefix is also to blame, that's unusual and will likely conflict with several system files, so it seems very much unadvisable to install to `/usr`.

**Comment by TastyPi** [comment]
If you look in the log file it's actually complaining about missing Perl modules, e.g. `Can't locate FindBin.pm in @INC`. I did `dnf install perl` to get all perl modules, but you could also install just the necessary modules if you want.

**Reply by TastyPi** [reply]
I decided to see what the minimum packages were, I think you only need `perl-FindBin` and `perl-File-Compare`

**Reply by nickmarden** [reply]
Maybe I'm misunderstanding, but I feel like this response misses the real issue. `FindBin` is part of the perl core.

Either the perl core should be installed as a dependency of `rbenv`, or the instructions should tell users that they need to install perl. I believe that neither is the case right now.

See https://github.com/openssl/openssl/issues/13761

**Reply by mislav** [reply]
Please note that the error from the full build log linked to by the OP doesn't not indicate a problem with perl modules. Furthermore, `perl` is in no way a dependency of ruby-build nor rbenv.

I don't know what the original problem was, but Ruby 2.6 is now EOL, so there is no point in trying to debug it. If someone is experiencing problems installing newer (supported) Ruby versions under Fedora, please open a new discussion. Thanks!
